samples/spring-transfer-workflow/src/main/java/com/example/transfer/TransferWorkflowEntity.java

```diff
@@ -0,0 +1,181 @@
+@EntityKey("transferId") // <3>
+@RequestMapping("/transfer/{transferId}") // <4>
+public class TransferWorkflowEntity extends WorkflowEntity<TransferState> { // <1>
+  // end::class[]
+
+  // tag::start[]
+  public record Withdraw(String from, int amount) { // <4>
+  }
+
+  // end::start[]
+
+  // tag::definition[]
+  public record Deposit(String to, int amount) { // <4>
+  }
+
+  // end::definition[]
+
+
+  private static final Logger logger = LoggerFactory.getLogger(TransferWorkflowEntity.class);
+
+  private final String fraudDetectionStepName = "fraud-detection";
+
+  // tag::definition[]
+  private final String withdrawStepName = "withdraw"; // <1>
+  private final String depositStepName = "deposit"; // <1>
+
+  // end::definition[]
+
+  final private KalixClient kalixClient;
+
+  public TransferWorkflowEntity(KalixClient kalixClient) {
+    this.kalixClient = kalixClient;
+  }
+
+  // tag::definition[]
+  @Override
+  public Workflow<TransferState> definition() {
+    Step withdraw =
+        step(withdrawStepName) // <1>
+            .call((Withdraw cmd) -> kalixClient.patch("/wallet/" + cmd.from() + "/withdraw/" + cmd.amount(), String.class)) // <2>
+            .andThen(this::moveToDeposit);
+
+    Step deposit =
+        step(depositStepName) // <1>
+            .call((Deposit cmd) -> kalixClient.patch("/wallet/" + cmd.to() + "/deposit/" + cmd.amount(), String.class)) // <5>
+            .andThen(this::finishWithSuccess);
+
+    return workflow() // <7>
+        .addStep(withdraw)
+        .addStep(deposit);
+  }
+
+  private Effect.TransitionalEffect<Void> moveToDeposit(String response) {
+    TransferState transferState = currentState();
+    TransferState successfulWithdrawal = transferState.asSuccessfulWithdrawal();
+    Deposit depositInput = new Deposit(transferState.transfer().to(), transferState.transfer().amount()); // <4>
+    return effects()
+        .updateState(successfulWithdrawal)
+        .transition(depositInput, depositStepName); // <3>
+  }
+
+  private Effect.TransitionalEffect<Void> finishWithSuccess(String response) {
+    TransferState completed = currentState().asCompleted();
+    return effects()
+        .updateState(completed)
+        .end(); // <6>
+  }
+  // end::definition[]
+
+  // tag::start[]
+  @PutMapping
+  public Effect<Message> startTransfer(@RequestBody Transfer transfer) {
+    if (transfer.amount() <= 0) {
+      return effects().error("transfer amount should be greater than zero"); // <1>
+    } else if (currentState() == null) {
+
+      TransferState initialState = new TransferState(transfer); // <2>
+
+      Withdraw withdraw = new Withdraw(transfer.from(), transfer.amount());
+
+      return effects()
+          .updateState(initialState) // <3>
+          .transition(withdraw, withdrawStepName) // <4>
+          .thenReply(new Message("transfer started")); // <5>
+
+    } else {
+      return effects().reply(new Message("transfer already started")); // <6>
```

Comment:
I think this should be an `error`. Not essential to explain workflows though. 

Also, I think we should start with guard clauses and then the meat, for example:

```java
if (transfer.amount() <= 0) {
  return effects().error("transfer amount should be greater than zero");
} else if (currentState() != null) {
  return effects().error("transfer already started"));
} else {
  //... start the workflow now
}

```